cmd/metrics.go

```diff
@@ -202,4 +203,26 @@ func metricsHandler() http.Handler {
 				ErrorHandling: promhttp.ContinueOnError,
 			}),
 	)
+
+}
+
+// AuthMiddleware checks if the bearer token is valid and authorized.
+func AuthMiddleware() func(http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			config := globalServerConfig.GetPrometheusConfig()
+			if !config.Enable {
+				writeErrorResponse(context.Background(), w, errorCodes.ToAPIErr(ErrPrometheusDisabled), r.URL, guessIsBrowserReq(r))
+				return
+			}
+			if config.Auth == globalPrometheusJwtAuth {
+				_, _, authErr := webRequestAuthenticate(r)
+				if authErr != nil {
+					writeWebErrorResponse(w, authErr)
```

Comment:
We don't need to write JSON error response - these are not Web API operations, we should just write what prometheus client can consume as an error.